Mathlib/NumberTheory/JacobiSum/Basic.lean

```diff
@@ -0,0 +1,120 @@
+/-
+Copyright (c) 2024 Michael Stoll. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Michael Stoll
+-/
+import Mathlib.NumberTheory.MulChar.Basic
+import Mathlib.Algebra.Module.BigOperators
+
+/-!
+# Jacobi Sums
+
+This file defines the *Jacobi sum* of two multiplicative characters `χ` and `ψ` on a finite
+commutative ring `R` with values in another commutative ring `R'`:
+
+`jacobiSum χ ψ = ∑ x : R, χ x * ψ (1 - x)`
+
+(see `jacobiSum`) and provides some basic results and API lemmas on Jacobi sums.
+
+## References
+
+We essentially follow
+* [K. Ireland, M. Rosen, *A classical introduction to modern number theory*
+   (Section 8.3)][IrelandRosen1990]
+
+but generalize where appropriate.
+
+This is based on Lean code written as part of the bachelor's thesis of Alexander Spahl.
+-/
+
+open BigOperators Finset
+
+/-!
+### Jacobi sums: definition and first properties
+-/
+
+section Def
+
+namespace Finset
+
+-- Is this useful enough to be non-private?
+private lemma sum_eq_sum_one_sub {R M : Type*} [Ring R] [Fintype R] [AddCommMonoid M] (f : R → M) :
```

Comment:
Maybe put the more complicated term on the LHS of the inequality?

Reply:
Good point. The proof doesn't even need to change...

Reply:
Actually, it even gets simpler!